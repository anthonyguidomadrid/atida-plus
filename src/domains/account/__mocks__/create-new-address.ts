export const createNewAddress = {
  redirect: 'some-path',
  salutation: 'string',
  firstName: 'string',
  lastName: 'string',
  address1: 'string',
  address3: 'string',
  zipCode: 'string',
  city: 'string',
  country: 'string',
  iso2Code: 'string',
  phone: 'string',
  isDefaultShipping: true,
  isDefaultBilling: true
}

export const createNewAddressPayload = {
  data: {
    type: 'customers',
    attributes: {
      salutation: 'string',
      firstName: 'string',
      lastName: 'string',
      address1: 'string',
      address3: 'string',
      zipCode: 'string',
      city: 'string',
      country: 'string',
      iso2Code: 'string',
      phone: 'string',
      isDefaultShipping: true,
      isDefaultBilling: true
    }
  }
} as const

export const UpdatedJWTAndRefreshTokens = {
  refreshToken: 'newRefreshToken',
  JWT: 'newJWT',
  expiresIn: '12345'
}

export const sprykerUpdatedJWTAndRefreshTokens = {
  data: {
    type: '',
    id: '',
    attributes: {
      tokenType: 'Bearer',
      expiresIn: '12345',
      accessToken: 'newJWT',
      refreshToken: 'newRefreshToken',
      idCompanyUser: ''
    },
    links: {
      self: ''
    }
  },
  links: {
    self: ''
  }
}
