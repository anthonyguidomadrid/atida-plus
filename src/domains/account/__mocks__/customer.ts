export const createCustomer = {
  salutation: 'Mr',
  firstName: 'Test',
  lastName: 'McTest',
  email: 'some@email.com',
  password: 'somePassword1!',
  confirmPassword: 'somePassword1!',
  acceptedTerms: true,
  emailNotification: true,
  phone: '654654654',
  company: null,
  surcharge: false,
  addresses: {
    addresses: [
      {
        salutation: 'Mr',
        firstName: 'Test',
        lastName: 'McTest',
        address1: 'Brugstraat 123',
        zipCode: '1011 AD',
        city: 'Amsterdam',
        iso2Code: 'PT',
        phone: '088273791298432',
        isDefaultShipping: true,
        isDefaultBilling: true
      }
    ]
  }
}

export const sprykerCustomer = {
  data: {
    type: 'customers',
    id: 'some-id',
    attributes: {
      salutation: 'Mr',
      firstName: 'Test',
      lastName: 'McTest',
      email: 'some@email.com',
      password: 'somePassword1!',
      confirmPassword: 'somePassword1!',
      acceptedTerms: true,
      emailNotification: true,
      company: null,
      surcharge: false
    }
  }
} as const

export const sprykerCustomerWithAddresses = {
  data: {
    ...sprykerCustomer,
    attributes: {
      ...sprykerCustomer.data.attributes
    },
    relationships: {
      addresses: {
        data: [
          { type: 'addresses', id: 'address-id-1' },
          { type: 'addresses', id: 'address-id-2' }
        ]
      }
    }
  },
  included: [
    {
      type: 'addresses',
      id: 'address-id-1',
      attributes: {
        salutation: 'some-salutation',
        firstName: 'some-name',
        lastName: 'some-last-name',
        address1: 'some-address',
        address2: 'some-address2',
        address3: 'some-address3',
        zipCode: 'some-zip-code',
        city: 'some-city',
        country: 'some-country',
        iso2Code: 'some-iso-code',
        company: 'some-company',
        phone: 'some-phone',
        isDefaultShipping: true,
        isDefaultBilling: true,
        isTaxExempt: false
      }
    },
    {
      type: 'addresses',
      id: 'address-id-2',
      attributes: {
        salutation: 'some-other-salutation',
        firstName: 'some-other-name',
        lastName: 'some-other-last-name',
        address1: 'some-other-address',
        address2: 'some-other-address2',
        address3: 'some-other-address3',
        zipCode: 'some-other-zip-code',
        city: 'some-other-city',
        country: 'some-other-country',
        iso2Code: 'some-other-iso-code',
        company: 'some-other-company',
        phone: 'some-other-phone',
        isDefaultShipping: false,
        isDefaultBilling: false,
        isTaxExempt: false
      }
    }
  ]
}
