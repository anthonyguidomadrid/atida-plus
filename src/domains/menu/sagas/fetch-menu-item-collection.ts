import type { SagaIterator } from 'redux-saga'
import { all, call, getContext, put, takeEvery } from 'redux-saga/effects'

import { createClient, logger } from '~helpers'
import {
  getMenuFailure,
  getMenuFulfill,
  getMenuRequest,
  getMenuSuccess,
  getMenuTrigger
} from '~domains/menu'
import { Menu } from '~domains/contentful'
import { getErrorMessage } from '~helpers/error'

const dispatchMenuItemCollectionFetchRequest = async (
  locale: string,
  menuTitle: string
): Promise<Menu> => {
  if (typeof window === 'undefined') {
    const {
      fetchMenuItemCollection
      // eslint-disable-next-line @typescript-eslint/no-var-requires
    } = require('../services/fetch-menu-item-collection')
    return fetchMenuItemCollection(locale, menuTitle)
  } else {
    const client = createClient({ locale })
    const response = await client.get<Menu>(
      `/api/submenu/get?menuTitle=${menuTitle}`
    )

    return response.data
  }
}

function* getMenuSaga({ payload }: ReturnType<typeof getMenuTrigger>) {
  const { menuTitle } = payload

  try {
    yield put(getMenuRequest())
    const locale: string = yield getContext('locale')
    const response: Menu = yield call(
      dispatchMenuItemCollectionFetchRequest,
      locale,
      menuTitle
    )
    yield put(getMenuSuccess(response))
  } catch (error) {
    logger.error(getErrorMessage(error))
    yield put(
      getMenuFailure({
        message: getErrorMessage(error)
      })
    )
  } finally {
    yield put(getMenuFulfill())
  }
}

export function* fetchMenuItemCollectionSaga(): SagaIterator {
  yield all([takeEvery(getMenuTrigger, getMenuSaga)])
}
