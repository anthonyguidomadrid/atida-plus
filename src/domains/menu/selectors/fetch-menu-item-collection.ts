import { createSelector } from '@reduxjs/toolkit'

import { RootState } from '~domains/redux'

const selectMenu = (state: RootState) => state.server.menu

export const selectMenuData = createSelector(
  selectMenu,
  menu => menu?.menu?.menu
)
