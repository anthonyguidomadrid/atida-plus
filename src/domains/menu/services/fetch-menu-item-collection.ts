import getConfig from 'next/config'

import { createClient, transformLocaleToUppercase } from '~helpers'
import { cache } from '~helpers/server-only/redisClient'
import { GetMenu, GetMenuQuery } from '~generated-graphql'
import { Menu, normalizeMenu } from '~domains/contentful'

if (typeof window !== 'undefined')
  throw new Error('This file should not be called client-side')

export const fetchMenuItemCollection = async (
  locale: string,
  menuTitle: string
): Promise<Menu | undefined> => {
  const { serverRuntimeConfig } = getConfig()

  const client = createClient({
    options: {
      baseURL: `${serverRuntimeConfig.cmsGraphQLHost}/spaces/${serverRuntimeConfig.cmsSpaceId}/environments/${serverRuntimeConfig.cmsEnvironmentId}`,
      headers: {
        Authorization: `Bearer ${serverRuntimeConfig.cmsToken}`
      },
      adapter: cache?.adapter
    },
    locale,
    interceptorOptions: {
      mode: 'contentful-graphql'
    }
  })

  const menuResponse = await client.post<{
    data: GetMenuQuery
  }>('', {
    query: GetMenu.loc?.source.body,
    variables: {
      locale: transformLocaleToUppercase(locale),
      menuTitle
    }
  })

  return normalizeMenu(menuResponse?.data?.data?.menuCollection ?? undefined)
}
