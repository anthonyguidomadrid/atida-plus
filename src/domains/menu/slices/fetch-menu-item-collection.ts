import { createSlice } from '@reduxjs/toolkit'
import type { PayloadAction } from '@reduxjs/toolkit'
import { Menu } from '~domains/contentful'
import { hydrate } from '~domains/redux/actions'
import { hasOwnProperty } from '~helpers'

export type MenuPayload = {
  menuTitle: string
}

type MenuState = {
  isLoading: boolean
  wasSuccess: boolean
  wasError: boolean
  error?: string
  menu?: Menu | null
}

const initialState: MenuState = {
  isLoading: false,
  wasSuccess: false,
  wasError: false,
  menu: null
}

export const fetchMenuItemCollectionSlice = createSlice({
  name: 'menu',
  initialState,
  reducers: {
    trigger(_state, _action: PayloadAction<MenuPayload>) {
      // no-op, used for triggering saga
    },
    request(state) {
      state.isLoading = true
      state.wasSuccess = false
      state.wasError = false
      delete state.error
    },
    success(state, action: PayloadAction<Menu>) {
      const data = action.payload
      if (typeof data !== 'undefined') state.menu = data
      state.wasSuccess = true
    },
    failure(state, action: PayloadAction<{ message: string }>) {
      const { message } = action.payload
      state.wasError = true
      state.error = message
    },
    fulfill(state) {
      state.isLoading = false
    }
  },
  extraReducers: builder => {
    builder.addCase(hydrate, (_state, action) => {
      if (
        typeof action.payload === 'object' &&
        hasOwnProperty(action.payload, 'menu') &&
        typeof action.payload.menu === 'object' &&
        hasOwnProperty(action.payload.menu, 'menu')
      ) {
        return action.payload.menu['menu'] as MenuState
      }
    })
  }
})
