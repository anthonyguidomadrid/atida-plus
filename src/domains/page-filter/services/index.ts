export * from './fetch-page-filter'
