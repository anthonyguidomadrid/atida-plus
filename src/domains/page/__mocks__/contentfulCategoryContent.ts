import { GetMenuQuery, PageQuery } from '~generated-graphql'
import { PageContent, PageType } from '../types'

export const contentfulCategoryPageRest = {
  total: 1,
  skip: 0,
  limit: 100,
  items: [
    {
      sys: {
        contentType: {
          sys: { type: 'Link', linkType: 'ContentType', id: 'page' }
        }
      },
      fields: {
        title: 'Digestion',
        slug: 'medicines/digestion',
        category: {
          sys: {
            contentType: {
              sys: { type: 'Link', linkType: 'ContentType', id: 'category' }
            }
          },
          fields: {
            title: 'Digestion',
            id: 'Medicines > Digestion',
            color: {
              sys: {
                contentType: {
                  sys: {
                    type: 'Link',
                    linkType: 'ContentType',
                    id: 'color'
                  }
                }
              },
              fields: { ref: 'category-beauty' }
            },
            image: {
              sys: {
                type: 'Asset'
              },
              fields: {
                title: 'Beauty Category Image',
                file: {
                  url:
                    '//images.ctfassets.net/7g2w796onies/3Pa3xwIiNIsKGQB2SE2KYy/9045bf905aba7e22e69054b4419c58a9/category-header-sample.png',
                  details: { size: 249696, image: { width: 845, height: 336 } },
                  fileName: 'category-header-sample.png',
                  contentType: 'image/png'
                }
              }
            }
          }
        }
      }
    }
  ]
}

export const contentfulCategoryPageGraphQL: { data: PageQuery } = {
  data: {
    pageCollection: {
      items: [
        {
          __typename: 'Page',
          title: 'Digestion',
          slugOne: 'medicines/digestion',
          slugTwo: 'medicines/digestion',
          contentBlocksCollection: {
            items: []
          },
          pageType: 'Category',
          referencedContent: {
            __typename: 'Category',
            title: 'Digestion',
            id: 'Medicines > Digestion',
            level: 1,
            color: {
              __typename: 'Color',
              ref: 'category-beauty'
            },
            image: {
              __typename: 'Asset',
              url:
                '//images.ctfassets.net/7g2w796onies/3Pa3xwIiNIsKGQB2SE2KYy/9045bf905aba7e22e69054b4419c58a9/category-header-sample.png'
            },
            linkedFrom: {
              categoryCollection: {
                items: [
                  {
                    __typename: 'Category',
                    id: 'medicines',
                    linkedFrom: {
                      categoryCollection: {
                        items: [
                          { __typename: 'Category', id: 'medicines/digestion' }
                        ]
                      }
                    }
                  }
                ]
              }
            }
          },
          contentfulMetadata: {
            tags: [{ id: 'country-pt' }]
          }
        }
      ]
    }
  }
}

export const categoryPagePOP: PageContent = {
  type: PageType.POP,
  title: 'Digestion',
  slug: 'medicines/digestion',
  allSlugs: {
    'pt-pt': 'medicines/digestion'
  },
  isCampaignPage: false,
  contentBlocks: [],
  referencedContent: {
    color: 'category-beauty',
    title: 'Digestion',
    id: 'Medicines > Digestion',
    level: 1,
    image: {
      url:
        '//images.ctfassets.net/7g2w796onies/3Pa3xwIiNIsKGQB2SE2KYy/9045bf905aba7e22e69054b4419c58a9/category-header-sample.png'
    },
    path: {
      id0: 'medicines',
      id1: 'Medicines > Digestion',
      title1: 'Digestion'
    }
  },
  category: {
    color: 'category-beauty',
    id: 'Medicines > Digestion',
    title: 'Digestion',
    level: 1,
    image: {
      url:
        '//images.ctfassets.net/7g2w796onies/3Pa3xwIiNIsKGQB2SE2KYy/9045bf905aba7e22e69054b4419c58a9/category-header-sample.png'
    },
    path: {
      id0: 'medicines',
      id1: 'Medicines > Digestion',
      title1: 'Digestion'
    }
  }
}

export const categoryPageCOP = {
  ...categoryPagePOP,
  type: PageType.COP,
  categoryCop: {
    level: 0,
    category: {
      ...categoryPagePOP.category,
      level: 0,
      subcategories: [
        { id: '1', url: '/sub1', title: 'sub1' },
        { id: '2', url: '/sub2', title: 'sub2' }
      ]
    },
    url: '/beauty',
    color: 'beauty-color'
  }
}

export const menuMock = {
  id: 'test-menu',
  title: 'testTitle',
  items: [
    {
      id: 'submenu1',
      title: 'Submenu title 1',
      url: 'test/url'
    },
    {
      id: 'submenu2',
      title: 'Submenu title 2',
      url: 'test/url'
    }
  ]
}

export const menuResponseMock: { data: { data: GetMenuQuery } } = {
  data: {
    data: {
      menuCollection: {
        __typename: 'MenuCollection',
        items: [{ ...menuMock, __typename: 'Menu' }]
      }
    }
  }
}
