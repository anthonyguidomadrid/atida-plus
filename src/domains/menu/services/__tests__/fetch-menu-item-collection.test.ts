/**
 * @jest-environment node
 */
import { fetchMenuItemCollection } from '~domains/menu/services/fetch-menu-item-collection'
import axios from 'axios'
import { menuResponseMock } from '~domains/page/__mocks__/contentfulCategoryContent'

describe(fetchMenuItemCollection, () => {
  it('creates a client and passes the locale and menu title to the request', async () => {
    ;(axios.post as jest.Mock).mockResolvedValue({})

    await fetchMenuItemCollection('pt-PT', 'testTitle')
    expect(axios.post).toHaveBeenCalledWith('', {
      query: expect.any(String),
      variables: { locale: 'pt-PT', menuTitle: 'testTitle' }
    })
  })

  it('returns a normalized response', async () => {
    ;(axios.post as jest.Mock).mockResolvedValue(menuResponseMock)

    const response = await fetchMenuItemCollection('pt-PT', 'testTitle')
    expect(axios.post).toHaveBeenCalled()
    expect(response).toMatchSnapshot()
  })
})
