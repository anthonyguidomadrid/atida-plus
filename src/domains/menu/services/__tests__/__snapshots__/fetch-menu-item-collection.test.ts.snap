// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchMenuItemCollection returns a normalized response 1`] = `
Object {
  "id": "test-menu",
  "title": "testTitle",
}
`;
